src/app/settings/emergency-access-takeover.component.ts

```diff
@@ -45,25 +50,29 @@ export class EmergencyAccessTakeoverComponent extends ChangePasswordComponent im
             platformUtilsService, policyService);
     }
 
-    // tslint:disable-next-line
-    async ngOnInit() { }
+    async ngOnInit() {
+        this.takeoverResponse = await this.apiService.postEmergencyAccessTakeover(this.emergencyAccessId);
+        if (this.takeoverResponse.policy != null && this.takeoverResponse.policy.length > 0)
+        {
+            const policy = this.takeoverResponse.policy.map((policyResponse: PolicyResponse) => new Policy(new PolicyData(policyResponse)));
+            this.enforcedPolicyOptions = await this.policyService.getMasterPasswordPolicyOptions(policy);
+        }
+     }
 
     async submit() {
         if (!await this.strongPassword()) {
             return;
         }
 
-        const takeoverResponse = await this.apiService.postEmergencyAccessTakeover(this.emergencyAccessId);
-
-        const oldKeyBuffer = await this.cryptoService.rsaDecrypt(takeoverResponse.keyEncrypted);
+        const oldKeyBuffer = await this.cryptoService.rsaDecrypt(this.takeoverResponse.keyEncrypted);
         const oldEncKey = new SymmetricCryptoKey(oldKeyBuffer);
 
         if (oldEncKey == null) {
             this.toasterService.popAsync('error', this.i18nService.t('errorOccurred'), this.i18nService.t('unexpectedError'));
             return;
         }
 
-        const key = await this.cryptoService.makeKey(this.masterPassword, this.email, takeoverResponse.kdf, takeoverResponse.kdfIterations);
+        const key = await this.cryptoService.makeKey(this.masterPassword, this.email, this.takeoverResponse.kdf, this.takeoverResponse.kdfIterations);
```

Comment:
Do we know what happens if the user we takeover managed to change their encryption key between the modal opening and the submit event being triggered? The modal could potentially be opened quite a bit longer than before.

Reply:
That's a good question Oscar.... @eliykat ?

Reply:
Good question! I just tested this situation and it didn't stop the takeover from going through or cause any unexpected behaviour. That said, I want to make sure I'm understanding the crypto-magic behind takeover before waving this through, so I'll have a chat with Oscar.

If it is a problem - or if we're just not comfortable with having stale `takeoverResponse` data client-side as a matter of best practice - the alternative would be to separate out the api calls, so that we get the grantor policies on `ngOn Init()` and move the `takeoverResponse` call back into `submit()`.

Reply:
Update: it's not really an issue if the grantor changes their master password while this modal is open, but if they rotate their encryption key, it has some... extremely bad outcomes. I have successfully nuked one (1) local testing vault to verify this.

I'll separate out the api calls as I mentioned above.